doc/Reference/Reactive/state.md

```diff
@@ -164,3 +213,167 @@ This gives you the ability to update a _state_, including the metadata.
+    
+```csharp
+public async ValueTask SetSliderMiddle(CancellationToken ct = default)
+{        
+    await SliderValue.SetAsync(50, ct);
+}
+```
+    
+### Subscribing to changes
+    
+ The `ForEach` enables executing a callback each time the value of the `IState<T>` is updated.
+    
+This extension-method takes a single parameter which is a async callback that takes two parameters. The first parameter is of type `T?`, where `T` is type of the `IState`, and represents the new value of the state. The second parameter is a `CancellationToken` which can be used to cancel a long running action.
+        
+```csharp
+public static IDisposable ForEach<T>(this IState<T> state, Func<T?, CancellationToken, Task> action);
+ ```
+
+For example:    
+```csharp
+   public partial record Model
+   {
+       public IState<string> MyState => ...
+   
+       public async ValueTask EnableChangeTracking()
+       {
+           MyState.ForEach(PerformAction);
+       }
+   
+       public async ValueTask PerformAction(string item, CancellationToken ct)
+       {
+           ...
+       }
+   }
+   ```
+    
+Additionally, the `ForEach` method can be set using the Fluent API:
+    
+   ```csharp
+   public partial record Model
+   {
+       public IState<string> MyState => State.Value(this, "Initial value")
+                                             .ForEach(PerformAction);
+   
+       public async ValueTask PerformAction(string item, CancellationToken ct)
+       {
+           ...
+       }
+   }
+   
+   ```
+    
+   ### Commands
+    
+   Part of the MVUX toolbox is the automatic generation of Commands.
+   In the `IncrementSlider` example [we've just used](#change-data-of-a-state), a special asynchronous Command will be generated that can be used in the View by a `Button` or other controls:
+    
+   Let's modify the XAML [above](#how-to-bind-the-view-to-a-state) with the following:
+    
+   ```xml
+           ...
+           <TextBlock Text="Set state value:"/>
+           <Slider Value="{Binding SliderValue, Mode=TwoWay}" />
+   
+           <Button Content="Increment slider" Command="{Binding IncrementSlider}" />
+   
+       </StackPanel>
+   </Page>
+   ```
+    
+   When pressing the _Increment slider_ button, the generated `IncrementSlider` command will be executed invoking the `IncrementSilder` method on the Model resulting in an incrementation of the value.
+    
+   This is what the result will look like:
+    
+   ![A video that demonstrates the effect of the recent updates applied to the slider-app](/Learn/Mvux/Assets/SliderApp-2.gif)
+    
+   The source code for the sample app can be found [here](https://github.com/unoplatform/Uno.Samples/tree/master/UI/MvuxHowTos/SliderApp).
```

Comment:
use a permalink to the folder: 
<img width="1684" alt="image" src="https://github.com/user-attachments/assets/dd3242f8-f4d2-44e3-b756-fca5fd5ab1c6">